subsys/bluetooth/mesh/net_keys.h

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SUBSYS_BLUETOOTH_MESH_NET_KEYS_H_
+#define ZEPHYR_SUBSYS_BLUETOOTH_MESH_NET_KEYS_H_
+
+#include <stdint.h>
+#include <sys/types.h>
+#include <sys/slist.h>
+#include <net/buf.h>
+
+#define BT_MESH_NET_FLAG_KR       BIT(0)
+#define BT_MESH_NET_FLAG_IVU      BIT(1)
+
+#define BT_MESH_KR_NORMAL         0x00
+#define BT_MESH_KR_PHASE_1        0x01
+#define BT_MESH_KR_PHASE_2        0x02
+#define BT_MESH_KR_PHASE_3        0x03
+
+/** Which of the two subnet.keys should be used for sending. */
+#define SUBNET_KEY_TX_IDX(sub) ((sub)->kr_phase == BT_MESH_KR_PHASE_2)
+
+struct bt_mesh_net_rx;
+enum bt_mesh_key_evt;
+
+/** Network message encryption credentials */
+struct bt_mesh_net_cred {
+	uint8_t nid;         /* NID */
+	uint8_t enc[16];     /* EncKey */
+	uint8_t privacy[16]; /* PrivacyKey */
+};
+
+/** Subnet instance. */
+struct bt_mesh_subnet {
```

Comment:
`net_keys.h` seems a bit of an odd choice for the `struct bt_mesh_subnet` definition? `struct bt_mesh_subnet_keys` seems appropriate here though.

Reply:
net_keys.c now holds the subnet array. Perhaps the most appropriate action is to rename these files to subnet.c and subnet.h?

Reply:
That would be fine with me